wgpu/src/renderer/widget/scrollable.rs

```diff
@@ -16,15 +17,57 @@ fn scrollbar_bounds(bounds: Rectangle) -> Rectangle {
     }
 }
 
+fn scroller_bounds(
+    bounds: Rectangle,
+    content_bounds: Rectangle,
+    background_bounds: Rectangle,
+    offset: u32,
+) -> Rectangle {
+    let ratio = bounds.height / content_bounds.height;
+    let scrollbar_height = bounds.height * ratio;
+    let y_offset = offset as f32 * ratio;
+
+    Rectangle {
+        x: background_bounds.x + f32::from(SCROLLBAR_MARGIN),
+        y: background_bounds.y + y_offset,
+        width: background_bounds.width - f32::from(2 * SCROLLBAR_MARGIN),
+        height: scrollbar_height,
+    }
+}
+
 impl scrollable::Renderer for Renderer {
-    fn is_mouse_over_scrollbar(
+    fn scrollbar_bounds(
+        &self,
+        bounds: Rectangle,
+        content_bounds: Rectangle,
+        offset: u32,
+    ) -> (Rectangle, Rectangle) {
+        let background_bounds = background_bounds(bounds);
+        let scroller_bounds =
+            scroller_bounds(bounds, content_bounds, background_bounds, offset);
+
+        (background_bounds, scroller_bounds)
+    }
+
+    fn scrollbar_grab(
         &self,
         bounds: Rectangle,
         content_bounds: Rectangle,
+        background_bounds: Rectangle,
+        scroller_bounds: Rectangle,
         cursor_position: Point,
-    ) -> bool {
-        content_bounds.height > bounds.height
-            && scrollbar_bounds(bounds).contains(cursor_position)
+    ) -> Option<ScrollbarGrab> {
+        if content_bounds.height > bounds.height
+            && background_bounds.contains(cursor_position)
+        {
+            Some(if scroller_bounds.contains(cursor_position) {
+                ScrollbarGrab::Scroller
+            } else {
+                ScrollbarGrab::Background
+            })
+        } else {
+            None
+        }
```

Comment:
I think this method could also live in `iced_native` entirely. There is nothing renderer-specific here, I think.

This should also allow us to not expose the `ScrollbarGrab` type in `iced_native`.